kornia/contrib/models/sam/model.py

```diff
@@ -172,8 +174,20 @@ def from_config(config: SamConfig) -> Sam:
         else:
             raise NotImplementedError('Unexpected config. The model_type should be provide or the encoder configs.')
 
-        if config.checkpoint:
-            model.load_checkpoint(config.checkpoint)
+        checkpoint = config.checkpoint
+        if config.pretrained:
+            if checkpoint is None:
+                checkpoint = {
+                    SamModelType.vit_b: "https://dl.fbaipublicfiles.com/segment_anything/sam_vit_b_01ec64.pth",
+                    SamModelType.vit_l: "https://dl.fbaipublicfiles.com/segment_anything/sam_vit_l_0b3195.pth",
+                    SamModelType.vit_h: "https://dl.fbaipublicfiles.com/segment_anything/sam_vit_h_4b8939.pth",
+                    SamModelType.mobile_sam: "https://github.com/ChaoningZhang/MobileSAM/raw/master/weights/mobile_sam.pt",
```

Comment:
shouldn't we use the permanent link instead?

https://github.com/ChaoningZhang/MobileSAM/blob/01ea8d0f5590082f0c1ceb0a3e2272593f20154b/weights/mobile_sam.pt